server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java

```diff
@@ -317,7 +317,8 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         Randomness.shuffle(candidateNodes);
 
         final Assignment assignment = persistentTasksExecutor.getAssignment(taskParams, candidateNodes, currentState);
-        assert (assignment == null || isNodeShuttingDown(currentState, assignment.getExecutorNode()) == false) :
+        assert assignment != null : "getAssignment() should always return an Assignment object, containing a node or a reason why not";
+        assert (assignment.getExecutorNode() == null || isNodeShuttingDown(currentState, assignment.getExecutorNode()) == false) :
```

Comment:
@dakrone and @henningandersen you may wish to review this change (just these 3 lines not the whole PR), as this change is in core code related to persistent tasks and the node shutdown API.

I am pretty sure that the idea was that `getAssignment` never returns `null`, but shout if you disagree.

I was getting NPEs from the original assertion when `assignment` was not `null` but `assignment.getExecutorNode()` was `null`.

Reply:
That change LGTM (sorry for the delayed response).